src/snowflake/connector/connection.py

```diff
@@ -1541,3 +1560,28 @@ def _all_async_queries_finished(self) -> bool:
             not self.is_still_running(self.get_query_status(q)) for q in queries
         )
         return all(finished_async_queries)
+
+    def _log_telemetry_imported_packages(self) -> None:
+        if self._log_imported_packages_in_telemetry:
+            # filter out duplicates caused by submodules
+            # and internal modules with names starting with an underscore
+            imported_modules = {
+                k.split(".", maxsplit=1)[0]
+                for k in sys.modules.keys()
+                if not k.startswith("_")
+            }
+            ts = get_time_millis()
+            self._log_telemetry(
+                TelemetryData(
+                    generate_telemetry_data(
+                        from_dict={
+                            TelemetryField.KEY_TYPE.value: TelemetryField.IMPORTED_PACKAGES.value,
+                            TelemetryField.KEY_SOURCE.value: self.application
+                            if self.application
+                            else CLIENT_NAME,
```

Comment:
Do we need to manually feed the `source` key-value pair into each of the telemetry calls, @sfc-gh-aling ?
If so then should this not be part of `generate_telemetry_data`?

Reply:
We should have `source`. `generate_telemetry_data` doesn't set it https://github.com/snowflakedb/snowflake-connector-python/blob/2c3f6d59cb824c568de15d5c646cddf35ebe8a3e/src/snowflake/connector/telemetry.py#L196-L215, and when we log job data, we do set `source`. https://github.com/snowflakedb/snowflake-connector-python/blob/2c3f6d59cb824c568de15d5c646cddf35ebe8a3e/src/snowflake/connector/cursor.py#L1221

Reply:
in my previous [fix PR](https://github.com/snowflakedb/snowflake-connector-python/pull/1231/files), I didn't realize that `source` could be part of the basic data for telemetry. I thought it is an operation based item as the exception telemetry does not include the `source` information.

but looks like `source` could/should be part of the basic information for those non-oob telemetry data.

let's add it?

Reply:
we could have a separate jira for the change.

Reply:
Yes, I think we should add it! Adam, can I merge this PR first, then you can modify my commit?

Reply:
Yes, let's have Adam's PR update this function call!